### PR TITLE
fix duration

### PR DESCRIPTION
**Pull request by fabersky**
screenshot 2016-03-03 a las 12 43 29

### CODE REVIEW

**Comment by JonRowe** [comment]
can you add a test showing what you want to change please

**Comment by fabersky** [comment]
As shown in the picture, "5 minutes 60 seconds" should be "6 minutes"

**Comment by myronmarston** [comment]
Can you add a new spec for the case of 360 seconds, showing that it returns "6 minutes" instead of "5 minutes 60 seconds"?  Without a new spec, this behavior could regress (since we rely so heavily on our spec suite).

**Comment by fabersky** [comment]
@myronmarston done!

**Comment by myronmarston** [comment]
Thanks, @fabersky.  I tried your spec locally and it passes but when I revert your fix (changing `duration >= 60` back to `duration > 60`) it still passes, so the spec you've written doesn't properly expose the issue.  I'm not sure why yet.

**Comment by myronmarston** [comment]
So, if I change the spec to:

``` ruby
        expect(helper.format_duration(359.999999999999)).to eq("6 minutes 0 seconds")
```

...then it fails as it should:

```
     Failure/Error: expect(helper.format_duration(359.999999999999)).to eq("6 minutes 0 seconds")

       expected: "6 minutes 0 seconds"
            got: "5 minutes 60 seconds"

       (compared using ==)
```

...but changing `duration > 60` to `duration >= 60` does nothing to fix the issue.  It appears to be a rounding issue: since `359.9999999999999` is less than 6 minutes, `5 minutes` is chosen for the minutes part, and then `59.9999999999` is rounded up to 60 seconds.

On a side note, this is why we always ask for a spec: without one, you might not be fixing the issue you think you are.

Want to take a stab at fixing it, @fabersky?

**Comment by fabersky** [comment]
Hi @myronmarston did you have a look at the failed tests in travis?

File: lib/rspec/core/formatters/helpers.rb [hunk 1]
```diff
@@ -34,6 +34,9 @@ def self.format_duration(duration)
             seconds = duration - minutes * 60
 
             "#{pluralize(minutes, 'minute')} #{pluralize(format_seconds(seconds, precision), 'second')}"
+          elsif duration % 60 == 1 && duration >= 60
+            minutes = 1
+            "#{pluralize(minutes, 'minute')}"
```
Comment (markets) [inline lib/rspec/core/formatters/helpers.rb]: This could be:

```
pluralize(1, 'minute')
```

**Comment by fabersky** [comment]
thanks @markets

**Comment by myronmarston** [comment]
> Hi @myronmarston did you have a look at the failed tests in travis?

The failure on travis is due to test coverage.  The new lines you've added [here](https://github.com/rspec/rspec-core/pull/2187/files#diff-d25edd3642c7e6123bb09957d78d4ec3R37) are uncovered by any test.  (In fact, if you change the logic in that branch to `raise "boom"`, all tests still pass).

To see what the coverage is, run `bundle exec script/rspec_with_simplecov` and it runs w/ simplecov loaded so you can see the coverage report locally.

Anyhow, as I mentioned above, if you want a test exposing the bug you ran into, you'll need to use an expectation like:

``` ruby
expect(helper.format_duration(359.999999999999)).to eq("6 minutes 0 seconds")
```

**Comment by myronmarston** [comment]
Closing in favor of #2208.

**Pull request closed without merging**